### ISSUE TITLE
on_success callback not called after job retried

### ISSUE THREAD

**Issue by tjwallace** [breamware/sidekiq-batch#32]
If a job inside of a batch is retried, the `on_complete` callback will be fired (before the failed job is retried) but the `on_success` callback will never fire, even after the failed job is retried successfully.

**Comment by rudrakshi2014** [comment]
> If a job inside of a batch is retried, the `on_complete` callback will be fired (before the failed job is retried) but the `on_success` callback will never fire, even after the failed job is retried successfully.

Did this issue got resolved? I'm still facing this issue.

**Comment by jrochkind** [comment]
Hi, if I underestand right, this even had a reproduction script (nice!), and is still not fixed, but  was closed just for being stale? 

If known bugs don't receive any maintainer attention and are closed for being stale without being fixed, the impression given to someone perusing the project to see it's maintenance status... is that the project is not really being maintained?

I'm not sure if this is an accurate impression or not? I am just perusing the project to get a sense of it's maintenance status. 

Reopening the issue, if it is indeed still a problem, might be in order?

**Comment by davidrichey** [comment]
I realized this is happening for us as well. I started to dig around why it was happening. From what I can tell right now when the `complete` callback gets called it will alway set the batch event key to `true` which prevents the `complete` callback from performing the `success` callback. I'm hoping to get some more time this weekend to play around and open a PR.

https://github.com/breamware/sidekiq-batch/blob/master/lib/sidekiq/batch.rb#L211
